### PR TITLE
Throw errors when .with is given missing keys

### LINKED ISSUE #18: Validating keys passed to .with
I notice that although `.new` validates the presence of all arguments, `.with` doesn't validate that all keys are provided (and defaults the attributes to nil). Is this intended behaviour? Is there are reason this is preferable? Seems like it would be easy to be tripped up by not specifying a value to `.with`.

If this behaviour is desirable, I'm happy to submit a PR

### PR DESCRIPTION
**Pull request by th3james**
Fixes #18
Couldn't bundle install with rcov on ruby >2, so I updated it to simplecov.

### CODE REVIEW

File: lib/values.rb [hunk 1]
```diff
@@ -23,6 +23,11 @@ def self.with(hash)
           raise ArgumentError.new("Unexpected hash keys: #{unexpected_keys}")
         end
 
+        missing_keys = self::VALUE_ATTRS - hash.keys
+        if missing_keys.any?
+          raise ArgumentError.new("Missing hash keys: #{missing_keys}")
```
Comment (tcrayford) [inline lib/values.rb]: would like this error to include the keys we got as well - more context is usually super helpful when dealing with most error messages.

**Comment by tcrayford** [comment]
one small error message fixup and this is good to go

**Comment by th3james** [comment]
Fixed :ok_hand:

**Comment by tcrayford** [comment]
released as 1.6.0: https://rubygems.org/gems/values

**Pull request merged**